File: Copilot/4.Patrón de Diseño Singleton/python/singleton.py
from flask import Flask, request

app = Flask(__name__)

@app.route('/saludo', methods=['GET'])
def saludo():
    nombre = request.args.get('nombre', 'amigo')
    return f"Hola, {nombre}!"

if __name__ == '__main__':
    app.run(debug=True)

File: Copilot/4.Patrón de Diseño Singleton/c++/singleton.cpp
// Ejemplo con Crow (framework C++ similar a Flask)
#include "crow_all.h"

int main() {
    crow::SimpleApp app;

    CROW_ROUTE(app, "/saludo")
    ([](const crow::request= req.url_params.get("nombre");
        std::string saludo = "Hola, ";
        saludo += (nombre ? nombre : "amigo");
        saludo += "!";
        return saludo;
    });

    app.port(18080).multithreaded().run();
}
